### PR TITLE
TOC, how-to/upgrade: update file name and TOC for 3.0 upgrade guide

### PR DESCRIPTION
**Pull request by dcalvin**
This PR updated the file name and TOC for 3.0 upgrade guide to be consistent with docs. 
Also updated some tool names for consistency.
PTAL @lilin90

### CODE REVIEW

File: TOC.md [hunk 1]
```diff
@@ -52,14 +52,14 @@
   + 扩容缩容
     - [使用 Ansible 扩容缩容](dev/how-to/scale/with-ansible.md)
   + 升级
-    - [升级至 TiDB 3.0](dev/how-to/upgrade/to-tidb-3.0.md)
+    - [升级至 TiDB 3.0](dev/how-to/upgrade/from-previous-version.md)
```
Comment (lilin90) [inline TOC.md]: This renaming affects the link in [this file](https://github.com/pingcap/docs-cn/blob/master/dev/how-to/upgrade/rolling-updates-with-ansible.md). Please update it.

File: TOC.md [hunk 2]
```diff
@@ -173,7 +173,7 @@
       + 从与 MySQL 兼容的数据库迁移数据
         - [从 Amazon Aurora MySQL 迁移数据](dev/how-to/migrate/from-aurora.md)
       - [Data Migration 故障诊断](dev/how-to/troubleshoot/data-migration.md)
```
Comment (lilin90) [inline TOC.md]:
```suggestion
      - [DM 故障诊断](dev/how-to/troubleshoot/data-migration.md)
```

**Comment by dcalvin** [comment]
PTAL again @lilin90 Thanks.

**Comment by dcalvin** [comment]
Need to close https://github.com/pingcap/docs-cn/pull/1385 for overlapping changes. @CaitinChen

**Pull request merged**